app2/src/lib/utils/queries.ts

```diff
@@ -29,36 +33,25 @@ export const fetchDecode = <S>(schema: Schema.Schema<S>, url: string) =>
     Effect.withLogSpan("fetchDecode"),
   )
 
-export type FetchDecodeGraphqlError = UnknownException | ParseError
+export type FetchDecodeGraphqlError = GraphQLError | Persistence.PersistenceError | ParseError
 /**
+ * TODO: Adjust calling convention to be `GraphQL` dependency injeciton.
  * @deprecated Migrate to `@unionlabs/sdk` query functions.
  */
-export const fetchDecodeGraphql = <S, E, D, V extends object | undefined>(
+export const fetchDecodeGraphql = <S, E, D, V extends Variables = Variables>(
   schema: Schema.Schema<S, E>,
   document: TadaDocumentNode<D, V>,
   variables?: V,
-): Effect.Effect<S, FetchDecodeGraphqlError, never> => {
-  const operationName = pipe(
-    document,
-    operationNamesFromDocumentNode,
-    A.head,
-    Option.getOrElse(() => "unknown"),
+): Effect.Effect<S, FetchDecodeGraphqlError, GraphQL> =>
+  Effect.andThen(
+    GraphQL,
+    ({ fetch }) =>
+      pipe(
+        fetch(new GraphQLRequest({ document, variables })),
```

Comment:
`GraphQL` service is only be responsible for querying, not schema transforms.

This may change if/when we migrate to declaring requests/results with more `Schema` classes.